lib/phoenix/live_dashboard/pages/ecto_stats_page.ex

```diff
@@ -3,63 +3,194 @@ defmodule Phoenix.LiveDashboard.EctoStatsPage do
   use Phoenix.LiveDashboard.PageBuilder
   import Phoenix.LiveDashboard.Helpers
 
-  @compile {:no_warn_undefined, [Decimal, EctoPSQLExtras]}
+  @compile {:no_warn_undefined, [Decimal, EctoPSQLExtras, {Ecto.Repo, :all_running, 0}]}
   @disabled_link "https://hexdocs.pm/phoenix_live_dashboard/ecto_stats.html"
+  @page_title "Ecto Stats"
 
   @impl true
-  def init(%{repo: nil}), do: {:ok, %{repo: nil, ecto_options: []}}
+  def init(%{repos: repos, ecto_psql_extras_options: ecto_options}) do
+    capabilities = for repo <- List.wrap(repos), do: {:process, repo}
+    repos = repos || :auto_discover
 
-  def init(%{repo: repo, ecto_psql_extras_options: ecto_options}),
-    do: {:ok, %{repo: repo, ecto_options: ecto_options}, process: repo}
+    {:ok, %{repos: repos, ecto_options: ecto_options}, capabilities}
+  end
 
   @impl true
-  def mount(_params, %{repo: repo, ecto_options: ecto_options}, socket) do
-    {:ok,
-     assign(socket,
-       repo: repo,
-       ecto_options: ecto_options,
-       info_module: info_module_for(repo)
-     )}
+  def mount(params, %{repos: repos, ecto_options: ecto_options}, socket) do
+    socket = assign(socket, ecto_options: ecto_options)
+
+    fun =
+      case repos do
+        :auto_discover ->
+          fn _ -> auto_discover() end
+
+        repos when is_list(repos) ->
+          &filter_repos_with_extra/1
+      end
+
+    case fun.(repos) do
+      {:ok, repos} ->
+        socket =
+          socket
+          |> assign(:repos, repos)
+          |> maybe_assign_nav_repo(params)
+
+        {:ok, socket}
+
+      {:error, error} ->
+        {:ok, assign(socket, :error, error)}
+    end
+  end
+
+  defp auto_discover do
+    with true <- function_exported?(Ecto.Repo, :all_running, 0),
+         [_ | _] = repos <- Ecto.Repo.all_running(),
```

Comment:
Those repos can be atoms or pids. Do we want to filter to be only atoms? I think we need a similar change on the Broadway one too.

Reply:
Also, don't we need to call this on the remote node instead using :rpc.call? 🤔

Reply:
In fact, it seems a lot of the code in here needs to be rewritten to use :rpc.call. But I guess we will learn what is broken once we ship plds. :)

Reply:
My mistake here :grimacing: I will give it a try. 

> Those repos can be atoms or pids.

Just curious: why this can be a `pid`?

Reply:
You can start an ecto repository without a name, then it will return only the pid I think.

Reply:
Oh, I see. Nice! I will try to simulate this scenario.